grails-app/services/org/pih/warehouse/inventory/OutboundStockMovementService.groovy

```diff
@@ -42,15 +42,12 @@ class OutboundStockMovementService {
         List<ShipmentType> shipmentTypes = params.list("shipmentType") ? params.list("shipmentType").collect{ ShipmentType.read(it) } : null
         Boolean isApprovalRequired = stockMovement?.origin?.isApprovalRequired()
 
-        def query = { isCountQuery ->
-            if (isCountQuery) {
-                projections {
-                    countDistinct "id"
-                }
-            } else {
-                setResultTransformer(CriteriaSpecification.DISTINCT_ROOT_ENTITY)
-            }
 
+        def stockMovementsIds = OutboundStockMovementListItem.createCriteria().list() {
```

Comment:
I'd suggest adding a comment above with the name of the ticket:
"OBPIH-5814: Because stock movement approvers could be filtered out from the requisition by given approvers in params, here we only return ids of stock movements and below we had to make another query to get stock movements with given id's, so that second query doesn't anymore cut any approvers"

or something similar.
I won't lie that even though I knew what you've done here, I had to think a bit when looking at the code of what is going on and how it is working.